src/test/compile-fail/empty-struct-unit-expr.rs

```diff
@@ -24,10 +24,10 @@ enum E {
 fn main() {
     let e2 = Empty2(); //~ ERROR expected function, found `Empty2`
     let e4 = E::Empty4();
-    //~^ ERROR `E::Empty4` is being called, but it is not a function
+    //~^ ERROR expected a function, but `E::Empty4` is `E`
```

Comment:
I find this somewhat more confusing than before, I have to say, though I agree the old message is not great. It seems like it'd be great to say something that suggests what people ought to do more clearly.

Perhaps "enum variant `E::Empty4` is declared without arguments and cannot be called like a function"?